libraries/chain/include/eosio/chain/types.hpp

```diff
@@ -426,6 +424,31 @@ namespace chainbase {
       v.resize_and_fill(size.value, [&s](char* buf, std::size_t sz) { s.read(buf, sz); });
       return s;
    }
+
+   // chainbase::shared_cow_vector
+   template<typename DataStream, typename T> inline DataStream& operator<<( DataStream& s, const chainbase::shared_cow_vector<T>& v )  {
+      FC_ASSERT( v.size() <= MAX_SIZE_OF_BYTE_ARRAYS );
+      fc::raw::pack( s, fc::unsigned_int((uint32_t)v.size()));
+      const auto* data = v.data();
+      auto size = v.size();
+      for (std::size_t i = 0; i < size; ++i) {
+         fc::raw::pack(s, &data[i]);
+      }
+      return s;
+   }
+
+   template<typename DataStream, typename T> inline DataStream& operator>>( DataStream& s, chainbase::shared_cow_vector<T>& v )  {
+      fc::unsigned_int size;
+      fc::raw::unpack( s, size );
+      FC_ASSERT( size.value <= MAX_SIZE_OF_BYTE_ARRAYS );
+      FC_ASSERT( v.size() == 0 );
+      v.resize_and_fill(size.value, [&](auto* data, std::size_t sz) { 
+         for (std::size_t i = 0; i < sz; ++i) {
+            fc::raw::unpack(s, &data[i]);
```

Comment:
Shouldn't this be `fc::raw::unpack(s, data[i]);`

Reply:
Yes, good catch, thanks. There was also the issue that pack/unpack were not defined for shared?string/shared_vector. I just fixed this.